### PR TITLE
Allow getBuildAllTask() to return non-swift task

### PR DESCRIPTION
**Pull request by adam-fowler**
This allows the default build task to be something other than a swift process eg CMake.
The default build task is triggered by the test explorer and the background compilation

### CODE REVIEW

**Review by 0xTim — APPROVED** [review]

File: src/SwiftTaskProvider.ts [hunk 1]
```diff
@@ -128,26 +128,31 @@ export function createBuildAllTask(folderContext: FolderContext): vscode.Task {
...
-    let task = tasks.find(
-        task =>
-            task.name === `swift: ${buildTaskName}` &&
-            (task.execution as vscode.ProcessExecution).options?.cwd ===
-                folderContext.folder.fsPath &&
-            task.source === "Workspace"
-    );
+    const workspaceTasks = await vscode.tasks.fetchTasks();
+    let task = workspaceTasks.find(task => {
+        if (task.source !== "Workspace" || task.name !== `swift: ${buildTaskName}`) {
```
Comment (tristanlabelle) [inline src/SwiftTaskProvider.ts:139]: It looks like that won't work if our default build task is:

```json
		{
			"type": "cmake",
			"label": "App Build",
			"command": "build",
			"targets": [
				"all"
			],
			"problemMatcher": "$swiftc",
			"group": {
				"kind": "build",
				"isDefault": true
			},
			"detail": "Build App for selected configuration",
			"dependsOn": "Configure"
		},
```

Comment (adam-fowler) [reply src/SwiftTaskProvider.ts:139]: What, because the label is different?

Comment (adam-fowler) [reply src/SwiftTaskProvider.ts:139]: @tristanlabelle ?

Comment (fischman-bcny) [reply src/SwiftTaskProvider.ts:139]: @adam-fowler: @tristanlabelle is out for the week but yeah that's what I think he meant.
Is it not correct to search for a first kind:build/isDefault:true task?

Comment (adam-fowler) [reply src/SwiftTaskProvider.ts:139]: As I mentioned in the related issue, workspaces can have multiple packages in them. The default build task for a workspace might not compile the correct package. But I was thinking about and I can probably compare working directories as well to ensure a default task is the correct task for a package

Comment (adam-fowler) [reply src/SwiftTaskProvider.ts]: Maybe this change will work

**Review by tristanlabelle — APPROVED** [review]
I think that'll work!

File: src/SwiftTaskProvider.ts (hunk 1 above)
Comment (tristanlabelle) [reply src/SwiftTaskProvider.ts]: Sorry for the wait! I think the latest will work. Thanks!

**Comment by tristanlabelle** [comment]
I also now verified that it does work. Thanks!

**Comment by tristanlabelle** [comment]
When might we see a new release that includes this fix?

**Comment by adam-fowler** [comment]
Was thinking of doing one today/tomorrow. There are a couple of other things I'd like to get out

**Comment by adam-fowler** [comment]
And it is released (1.5)

**Pull request merged**